src/components/DropdownMenu/DropdownMenu.tsx

```diff
@@ -143,6 +135,13 @@ export const DropdownMenu: React.FC<Props> = ({
     }
   };
 
+  const handleOnClick: MouseEventHandler = (e) => {
+    focusIndex = refs.current.list.findIndex((ref) =>
+      ref.contains(e.target as HTMLElement),
+    );
```

Comment:
Finds which item had been clicked, and sets `focusIndex` accordingly so up and down arrow keys select the next or previous item